pkg/zkcertificate/certificate_uniswap.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2024 Galactica Network
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package zkcertificate
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/iden3/go-iden3-crypto/poseidon"
+	"github.com/shopspring/decimal"
+
+	"github.com/galactica-corp/guardians-sdk/internal/validation"
+)
+
+// UniswapInputs represent the input data for the Uniswap verification.
+type UniswapInputs struct {
```

Comment:
I am trying to make sense of the use case:

1. If the inputs contain data about the SwapVolume of the last Year and Half Year, wouldn't it make sense to also add a timestamp of when that measurement was taken?
2. What use does the Swap volume have? I could create absurdly high swap volumes artificially. I just need to create my own token and pair. Then I can swap back and forth with high volume. I think it would make more sense to measure something like tx fees paid during swaps or uniswap fees paid to a list of reputable pools.

Reply:
That's a good point. Let's discuss it with managers. Sincerely, I didn't think about it — just used the data provided to me

Reply:
1. Yes, it does. But don't we have certificate issue date as some default field? I think we should. Issue date + expiration date
2. good question, working on it

Reply:
2. I think we should filter the pools on the Guardian's side @HarmlessEvil ideally we shall only include those that are verified by uniswap

Reply:
> don't we have certificate issue date as some default field?

No, we don't